modules/core/src/test/java/org/apache/ignite/internal/processors/cache/consistency/AtomicCacheConsistencyTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.consistency;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import javax.cache.CacheException;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.CacheEntry;
+import org.apache.ignite.internal.processors.cache.distributed.near.consistency.IgniteConsistencyViolationException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Ideally, this test should be equal to {@link ImplicitTransactionalCacheConsistencyTest} except atomicy mode
+ * configuration.
+ *
+ * Implement repair on atomic caches to make this happen :)
+ */
+@RunWith(Parameterized.class)
+public class AtomicCacheConsistencyTest extends ImplicitTransactionalCacheConsistencyTest {
+    /**
+     *
+     */
+    protected static final Consumer<ReadRepairData> GET_CHECK_AND_FAIL =
```

Comment:
Can be private.

Reply:
Sure, but it protected to be similar to other closure fields.